codec-http2/src/main/java/io/netty/handler/codec/http2/Http2CodecUtil.java

```diff
@@ -272,10 +272,10 @@ private boolean allowNotificationEvent() {
         public ChannelPromise setSuccess(Void result) {
             if (allowNotificationEvent()) {
                 ++successfulCount;
-                if (successfulCount == expectedCount && doneAllocating) {
-                    promise.setSuccess(result);
-                    return super.setSuccess(result);
-                }
+            }
+            if (successfulCount == expectedCount && doneAllocating && !isDone()) {
```

Comment:
@Scottmitch +1 to having `doneAllocatingPromises` call `setSuccess(null)`, however I think it should just call `this` version so that we consolidate all of the promise logic in one place.  I feel that the new logic is actually clearer.